### DISCUSSION TITLE
Problem with time Update Field Value  Date Picker

### DISCUSSION THREAD

**Discussion by Bitello** [Budibase/budibase#9442] [Help] [answered]
Hi!

I'm having the following problem:
I need to use the "Update Field Value" Action to assign the current date/time to a screen attribute (I'm using BudiDB).

If I use Handlebars
{{ date }}
the return is
January 25, 2023
with the time at 00:00 when I use the "Update Field Value", but I need the time.
This is an example of a value assigned to the field:
{{ Form.Fields.dtBegin }} 2023-01-09T15:00:00.000Z

I tried to use JavaScript
return new Date().toLocaleString(); 01/25/2023 10:10:44 am
but the "Update Field Value" fails to assign the value to the field.

How do I proceed?

Grateful

**Comment by melohagan** [comment, marked as answer]
Hey @Bitello 

Could you share some screenshots please?

**Reply by Bitello** [reply]
Hi @melohagan.

Now that I noticed that even in the DatePicker dtInicio the time value is not correct.
![image](https://user-images.githubusercontent.com/22456825/214833845-8d12d49b-f0c2-4fe8-a1d8-3c85fcaeeb71.png)

Then...

I created two buttons.
One of them assigns the Handlebars {{ date }} to the dtFim field and the other returns a JavaScript.

In both there are two "Update field value" actions, the first Reset to default value 
![image](https://user-images.githubusercontent.com/22456825/214847721-544fd3a1-a7d7-4d43-9967-f8c4efece222.png)

and the second Set values.
![image](https://user-images.githubusercontent.com/22456825/214847808-1dc9adf5-8abe-4d6c-9011-60bb4ef4b2b2.png)

![image](https://user-images.githubusercontent.com/22456825/214850154-5e8ac00e-13b8-43e3-9406-6ed201ce74cb.png)

![image](https://user-images.githubusercontent.com/22456825/214850287-93e598ec-b094-40ec-b561-90f6806e41a7.png)

The result obtained is in the animation. When using Handlebars, the dtFim field correctly receives the date, but with the time at 00:00.

When clicking on the JavaScript button, the value is not assigned to the field (I tested it with several clicks in different orders and the same problem always occurs).

![gifntext-gif](https://user-images.githubusercontent.com/22456825/214846097-ecb2eddb-61ed-4dab-954b-40a3105d5555.gif)

**Reply by melohagan** [reply]
> Now that I noticed that even in the DatePicker dtInicio the time value is not correct.

You could try to tick the `Ignore time zones` setting and see if that helps.

The JavaScript might not be working because you are using locale string. It may need to be in the format: `2022-12-31T12:00:00.123`.
Or a Date object directly, i.e. just `return new Date()`

**Reply by Bitello** [reply]
Ignore time zones didn't work.

I changed the JavaScript to
return new Date().toString()
and it worked.


Thanks!
